### ISSUE TITLE
(Question) Clipboard cleanup

### ISSUE THREAD

**Issue by hardBSDk** [pop-os/cosmic-comp#616]
When the sender program of the clipboard is closed, the clipboard becomes empty.

Does it is the expected behavior? because I think it's a very important security measure to not allow untrusted programs to read old clipboard contents, like passwords.

**Comment by wiiznokes** [comment]
Well, I guess the current behavior that you describe is correct then ?
Since it's the role of the app to send the clipboard content, I think is enforced by the Wayland protocol

**Comment by Drakulix** [comment]
Yes this is a protocol measure. cosmic-comp would have to make an explicit copy of the clipboard contents, as transfer only takes place once two clients have negotiated a mine-type. As such a dead client can't and won't offer any clipboard contents, they are lost once it disconnects.

Users that don't like that behaviour an explicitly opt-in to using a clipboard manager.

**Comment by hardBSDk** [comment]
@Drakulix I love this protocol measure, thanks for making it the default.
